### PR TITLE
feat: V3 services in legacy endpoints

### PR DESCRIPTION
**Pull request by nsenave**
#563 

Eno API now uses V3 services for all (four) "DDI to Lunatic" endpoints 🎉 

- Some refactor in `core` to output Lunatic json or xml
- Replace redirection by v3 service call in DDI to Lunatic controllers

The V3 web-service should be ready to be used in combination with Pogues 🥳

### CODE REVIEW

File: eno-ws/src/main/java/fr/insee/eno/ws/controller/GenerationController.java [hunk 1]
```diff
@@ -17,12 +22,13 @@
 @Controller
 @RequestMapping("/questionnaire")
 @Slf4j
-@SuppressWarnings("unused")
 public class GenerationController {
 
+	private final V3ControllerUtils controllerUtils;
 	private final PassePlat passePlat;
```
Comment (davdarras) [inline eno-ws/src/main/java/fr/insee/eno/ws/controller/GenerationController.java:28]: english ?

File: eno-ws/src/main/java/fr/insee/eno/ws/controller/GenerationController.java [hunk 2]
```diff
@@ -73,6 +81,7 @@ public Mono<Void> generateFOQuestionnaire(
 					"For css parameters, separate style sheet by ','")
 	@PostMapping(value = "ddi-2-xforms",
 			produces = MediaType.APPLICATION_OCTET_STREAM_VALUE, consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+	@SuppressWarnings("unused")
```
Comment (davdarras) [inline eno-ws/src/main/java/fr/insee/eno/ws/controller/GenerationController.java:84]: if unused, maybe delete this method ?

File: eno-ws/src/main/java/fr/insee/eno/ws/controller/GenerationController.java [hunk 3]
```diff
@@ -48,6 +55,7 @@ public Mono<Void> generate(
 			description = "Generate a XSL-FO questionnaire from a DDI questionnaire using the FO parameters given.")
 	@PostMapping(value = "ddi-2-fo",
 			produces = MediaType.APPLICATION_OCTET_STREAM_VALUE, consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+	@SuppressWarnings("unused")
```
Comment (davdarras) [inline eno-ws/src/main/java/fr/insee/eno/ws/controller/GenerationController.java:58]: if unused, maybe delete this method

File: eno-ws/src/main/java/fr/insee/eno/ws/controller/GenerationController.java [hunk 4]
```diff
@@ -118,16 +128,60 @@ public Mono<Void> generateLunaticJsonQuestionnaire(
...
+		//
+		return controllerUtils.ddiToLunaticJson(ddiFile, parameters);
 	}
 
 	@Operation(
 			summary = "Generation of DDI questionnaire from Pogues xml questionnaire.",
 			description = "Generate a DDI questionnaire from a Pogues xml questionnaire.")
 	@PostMapping(value="poguesxml-2-ddi",
 			produces = MediaType.APPLICATION_OCTET_STREAM_VALUE, consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+	@SuppressWarnings("unused")
```
Comment (davdarras) [inline eno-ws/src/main/java/fr/insee/eno/ws/controller/GenerationController.java:184]: same here

File: eno-ws/src/main/java/fr/insee/eno/ws/controller/GenerationController.java [hunk 5]
```diff
@@ -139,6 +193,7 @@ public Mono<Void> generateDDIQuestionnaire(
 			description = "Generates a FODT (Open Document file) questionnaire from a DDI questionnaire.")
 	@PostMapping(value="ddi-2-fodt",
 			produces = MediaType.APPLICATION_OCTET_STREAM_VALUE, consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+	@SuppressWarnings("unused")
```
Comment (davdarras) [inline eno-ws/src/main/java/fr/insee/eno/ws/controller/GenerationController.java:196]: same here

File: eno-ws/src/main/java/fr/insee/eno/ws/service/DDIToLunaticService.java [hunk 6]
```diff
@@ -10,9 +11,18 @@
 @Service
```
Comment (davdarras) [inline eno-ws/src/main/java/fr/insee/eno/ws/service/DDIToLunaticService.java:11]: is there an interest as to use this class as a spring service ?

File: eno-ws/src/main/java/fr/insee/eno/ws/service/ParameterService.java [hunk 7]
```diff
@@ -25,4 +26,12 @@ public Mono<String> defaultParams() {
         }
     }
 
+    public Mono<String> defaultParams(EnoParameters.Context context, Format format) {
```
Comment (davdarras) [inline eno-ws/src/main/java/fr/insee/eno/ws/service/ParameterService.java:29]: is there an interest as to use this class as a spring service ?

**Review by davdarras — COMMENTED** [review]
Seems good to me, some little details ared added. Maybe some javadoc on methods (:D)

Is sonar project correctly configured ? I did not see code coverage

**Comment by nsenave** [comment]
> Seems good to me, some little details ared added. Maybe some javadoc on methods (:D)
> 
> Is sonar project correctly configured ? I did not see code coverage

True there's a lack of javadoc 🙊 

Sonar is not configured for the v3 branches, that's something I'd like to do soon

**Pull request merged**